### PR TITLE
crd: Add categories for cilium CRDs.

### PR DESCRIPTION
**Pull request by sayboras**
This commit is to add categories for cilium CRDs, so that users can
easily query cilium CRs with kubectl commands (e.g. kubectl get
cilium or kubectl get ciliumpolicy). Two categories (e.g. cilium and
ciliumpolicy) are added to start with.

Signed-off-by: Tam Mach <sayboras@yahoo.com>

Related slack [conversation](https://cilium.slack.com/archives/C2B917YHE/p1628855497104100)

```release-note
crd: Add categories for cilium CRDs
```

### CODE REVIEW

**Review by nathanjsweet — APPROVED** [review]
It would be nice if we could add some CI to ensure that future crds do not forget to add the categories field.

**Comment by sayboras** [comment]
> It would be nice if we could add some CI to ensure that future crds do not forget to add the categories field.

Sure, CI has evolved a lot since last time I take a look at it. Will do it in separate PR.

**Comment by sayboras** [comment]
I don't think the full CI is required, might need input here.

**Comment by aanm** [comment]
@sayboras I think a linter that checks for the existence of "categories" in that line should be enough?

**Comment by sayboras** [comment]
> I think a linter that checks for the existence of "categories" in that line should be enough?

Sure, I just add a simple check in second commit, let me know if this is what you have in mind. Manual testing was done by removing one category in ciliumclusterwidenetworkpolicies CRD.

```
$ make manifests
cd "./vendor/sigs.k8s.io/controller-tools/cmd/controller-gen" && \
go run ./... "crd:crdVersions=v1" paths="/home/tammach/go/src/github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2;/home/tammach/go/src/github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1" output:crd:artifacts:config="/tmp/tmp.WavdeFMNZO";
go run ./tools/categorycheck "/tmp/tmp.WavdeFMNZO"
2021/08/29 18:54:30 CRD categories missing for ciliumclusterwidenetworkpolicies.cilium.io
exit status 1
make: *** [Makefile:303: manifests] Error 1
```

**Review by qmonnet — APPROVED** [review]

File: tools/crdcheck/main.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			return err
+		}
+
+		crd, ok := obj.(*crdv1.CustomResourceDefinition)
+		if !ok {
+			return nil
+		}
+
+		for _, f := range allChecks {
+			if err = f(crd); err != nil {
```
Comment (sayboras) [inline tools/crdcheck/main.go:57]: currently, we have only one check, so i didn't use multi error here. Let's see how it's going.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
/test-1.16-netnext

**Review by aanm — APPROVED** [review]

**Pull request merged**